specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json

```diff
@@ -0,0 +1,2082 @@
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.KeyVault/locations/{location}/deletedVaults/{vaultName}/purge": {
+      "post": {
+        "tags": [
+          "Vaults"
+        ],
+        "operationId": "Vaults_PurgeDeleted",
+        "x-ms-long-running-operation": true,
+        "description": "Permanently deletes the specified vault. aka Purges the deleted Azure key vault.",
+        "parameters": [
+          {
+            "name": "vaultName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the soft-deleted vault."
+          },
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The location of the soft-deleted vault."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The vault is purged."
+          },
+          "202": {
+            "description": "Vault is being purged."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Purge a deleted vault": {
+            "$ref": "./examples/purgeDeletedVault.json"
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      }
+    },
+    "/subscriptions/{subscriptionId}/resources": {
+      "get": {
+        "tags": [
+          "Vaults"
+        ],
+        "operationId": "Vaults_List",
+        "description": "The List operation gets information about the vaults associated with the subscription.",
+        "parameters": [
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "resourceType eq 'Microsoft.KeyVault/vaults'"
+            ],
+            "x-ms-enum": {
+              "name": "VaultListFilterTypes",
+              "modelAsString": false
+            },
+            "description": "The filter to apply on the operation."
+          },
+          {
+            "name": "$top",
+            "in": "query",
+            "type": "integer",
+            "format": "int32",
+            "description": "Maximum number of results to return."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": true,
+            "x-ms-api-version": false,
+            "type": "string",
+            "enum": [
+              "2015-11-01",
+              "2022-02-01-preview"
```

Comment:
>2022 [](http://example.com/codeflow?start=15&length=4)

This is puzzling me here.
this does not look like the usual way of modeling apiVersion. - as an enum not modeled as string. But then it also does not look like a keyVaults API per se - it seems to be an **ARM api**, So... does it really support this api version??

[https://learn.microsoft.com/en-us/rest/api/resources/resources/list](https://learn.microsoft.com/en-us/rest/api/resources/resources/list)